test/suites/host-ip-change.robot

```diff
@@ -0,0 +1,233 @@
+    Set Global Variable    ${DEVICE}        ${device}
+
+    # Set the global variable for nmconnection_path
+    Set Global Variable    ${NMCONNECTION_PATH}    /etc/NetworkManager/system-connections/${device}.nmconnection
+
+Save Backup Default File
+    [Arguments]    ${file_path}
+    [Documentation]    Takes a backup of the given file and returns the path to the backup file.
+    ...    This keyword is meant to be used from a Setup step.
+    ${stdout}    ${rc}=    Execute Command
+    ...    cp ${file_path} ${file_path}.backup
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    ${rc}    0
+
+Restore Backup File
+    [Arguments]    ${backup_file}
+    [Documentation]    Restores the original file from the backup file.
+    ${stdout}    ${rc}=    Execute Command
+    ...    cp ${backup_file}.backup ${backup_file}
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    ${rc}    0
+
+Setup Suite
+    [Documentation]    Test suite setup
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig    # for readiness checks
+
+Rollback Configuration
+    [Arguments]    ${backup_file}
+    [Documentation]    Restore the original configuration from the backup file
+    Restore Backup File    ${backup_file}
+
+Add Node To Hosts File
+    [Arguments]    ${ip}    ${node_name}
+    [Documentation]    Add the node name and IP address to the /etc/hosts file
+    ${stdout}    ${rc}=    Execute Command
+    ...    echo "${ip} ${node_name}" | sudo tee -a /etc/hosts
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    ${rc}    0
+
+Get NodeName
+    [Documentation]    Get and Return nodename
+    ${nodeName}=    Run With Kubeconfig    oc get node --no-headers | awk '{print $1}'
+    Should Not Be Empty    ${nodeName}
+    [RETURN]    ${nodeName}
+
+Modify NetworkManager Configuration
+    [Arguments]    ${ip}    ${gateway}   ${device}
+    [Documentation]    Modify NetworkManager configuration using nmcli commands
+    ${stdout}   ${rc1}=    Execute Command    nmcli con modify ${device} ipv4.addresses ${ip}
+    ...    sudo=True    return_rc=True
+    ${stdout}   ${rc2}=    Execute Command    nmcli con mod ${device} ipv4.gateway ${gateway}
+    ...    sudo=True    return_rc=True
+    ${stdout}   ${rc3}=    Execute Command    nmcli con mod ${device} ipv4.method manual
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    ${rc1}     0
+    Should Be Equal As Integers    ${rc2}     0
+    Should Be Equal As Integers    ${rc3}     0
+
+Login MicroShift Host With New IP
+  [Arguments]  ${ip}
+  [Documentation]  Login to the MicroShift host via ssh and leave the connection open
+  ...  This keyword is meant to be used at the suite level. This ensures
+  ...  most tests already have an open connection. Any tests that will take
+  ...  action that disrupt that connection are responsible for restoring it.
+
+  Log  Host: ${ip}
```

Comment:
Arguments to keywords should be logged automatically

Reply:
Updated